cli/cycpp.py

```diff
@@ -1568,10 +1578,10 @@ def impl(self, ind="  "):
 
             if key in BUFFERS:  # buffer state, skip
                 continue
-            if 'derived_init' in info:  # derived state, skip
+            if 'internal' in info:  # derived state, skip
                 continue
 
-            opt = True if 'default' in info else False
+            opt = True if 'default' in info or 'derived_init' in info else False
```

Comment:
That's fine, the tests should have been easier (just adjust cycpp_tests.h).
Can you show me a build with an updated archetype that you're working on?

On Wed, Apr 1, 2015 at 4:32 PM, Robert Carlsen <notifications@github.com>
wrote:

> In cli/cycpp.py
> <https://github.com/cyclus/cyclus/pull/1110#discussion_r27614086>:
>
> >                  continue
> >
> > -            opt = True if 'default' in info else False
> > +            opt = True if 'default' in info or 'derived_init' in info else False
>
> You've convinced me it is not as bad as I initially thought. I'll change.
> I don't have time to do the tests though. I need to be moving on to other
> things.
> … <#14c76e625e819381_>
> On Wed, Apr 1, 2015 at 3:24 PM, Matthew Gidden <notifications@github.com>
> wrote: In cli/cycpp.py <
> https://github.com/cyclus/cyclus/pull/1110#discussion_r27613345>: >
> continue > > - opt = True if 'default' in info else False > + opt = True if
> 'default' in info or 'derived_init' in info else False In other words, the
> way derived_init works with this PR should be different than it works now.
> Cycamore facilities should have to add an 'internal' key to get the same
> behavior. — Reply to this email directly or view it on GitHub <
> https://github.com/cyclus/cyclus/pull/1110/files#r27613345>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cyclus/pull/1110/files#r27614086>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192